src/adam_bashforth.jl

```diff
@@ -1,57 +1,76 @@
-module Tmp
 ####################################
-# Explicit Adam-Bashforth solvers
+# Implicit Adam-Bashforth solvers
 ####################################
 # (Hairer & Wanner 1996, Vol I, p.357-358
-include("ODE.jl")
 
-##Fixed step, fixed order, multistep explicit method
 
-function ode_ab(F,y0, tspan,order ::Integer)
-    if !(2 <= order <= 4)
-        error("Currently only orders 2,3, and 4 are implemented for Adam-Bashforth method")
+##Function: ode_imp_ab(F,y0, tspan, order(optional))
+##order is set to 4 by default
+##Adam Bashforth is a fixed step, fixed order, multistep Implicit method
+function ode_imp_ab(F::Function,y0, tspan,order=4 ::Integer)
+    if (0 <= order <= 3)
+        coef = am_imp_coefficients3
+        b = ms_coefficients4
+    else
+        #calculating higher order coefficients for implicit Adam Multon method
+        coef = zeros(order+1, order+1)
+        coef[1:4,1:4] = am_imp_coefficients3
+        for s = 4:order
+            for j = 0:s
+                # Assign in correct order for multiplication below
+                #  (a factor depending on j and s) .* (an integral of a polynomial with -(-1:s-1), except -(j-1), as roots)
+                p_int = polyint(poly(diagm(-[-1:j - 2; j:s-1])))
+                coef[s+1, s+1 - j] = ((-1)^j / factorial(j)
+                               / factorial(s - j) * polyval(p_int, 1))
+            end
+        end
+        b = zeros(order, order)
+        b[1:4, 1:4] = ms_coefficients4
+        for s = 5:order
+            for j = 0:(s - 1)
+                # Assign in correct order for multiplication below
+                #  (a factor depending on j and s) .* (an integral of a polynomial with -(0:s), except -j, as roots)
+                p_int = polyint(poly(diagm(-[0:j - 1; j + 1:s - 1])))
+                b[s, s - j] = ((-1)^j / factorial(j)
+                               / factorial(s - 1 - j) * polyval(p_int, 1))
+            end
+        end
     end
 
-    stepsize = diff(tspan)
+    h = diff(tspan)
```

Comment:
In PR #49 standardizes the variable names to `t, dt, y, dy`.  Use that here too.